v_4_6_0/master_template.go

```diff
@@ -52,16 +52,19 @@ systemd:
     contents: |
       {{range .Content}}{{.}}
       {{end}}{{end}}
-  - name: set-certs-group-owner-permission-giantswarm.service
+  - name: set-giantswarm-user-ownership.service
     enabled: true
     contents: |
       [Unit]
-      Description=Change group owner for certificates to giantswarm
+      Description=Set required for giantswarm user ownership
       Wants=k8s-kubelet.service k8s-setup-network-env.service
       After=k8s-kubelet.service k8s-setup-network-env.service
       [Service]
       Type=oneshot
-      ExecStart=/bin/sh -c "find /etc/kubernetes/ssl -name '*.pem' -print | xargs -i  sh -c 'chown root:giantswarm {} && chmod 640 {}'"
+      # wait for kubelet dirs being created
+      ExecStartPre=/bin/sleep 30
```

Comment:
can we rather do check if the dir is already with some loop and sleep, this can easily fail on some envs ( like in china pulling image is slow and can definitely take more than 30s

Reply:
I don't think this service will be running before `k8s-kubelet.service` actually starts (goes through all StartPre and run ExecStart). And pulling is done in StartPre, so this ownership service can't be activated before

Reply:
anyway added loop